src/lavinmq/clustering/controller.cr

```diff
@@ -17,6 +17,22 @@ class LavinMQ::Clustering::Controller
     @id = clustering_id
     @advertised_uri = @config.clustering_advertised_uri ||
                       "tcp://#{System.hostname}:#{@config.clustering_port}"
+    @isr_key = "#{@config.clustering_etcd_prefix}/isr"
+  end
+
+  def password : String
```

Comment:
this can/will only be called by the leader? otherwise we have a racecondition in this method.

Reply:
Yes. The `Clustering::Client` is created by the `Clustering::Controller` and the password is passed as constructor argument. `Clustering::Client` doesn't know about neither `Etcd` nor `Clustering::Controller`.

Maybe `Clustering::Server` should be created by `Clustering::Controller` too. Then `#password` wont have to be public and `Clustering::Controller` handles it all.